### PR TITLE
fix: Preserve draft text when completing argument-taking slash commands

### PR DESCRIPTION
**Pull request by canvrno-oai**
This adds slash command completion behavior for argument-taking commands, where text after the partially typed command becomes inline arguments instead of being discarded. This addresses the workflow of drafting text first, moving to the start, and completing a slash command around that existing draft. Before this change, this workflow would remove all user-input text aside from the slash command, which can be frustrating if the user had just typed out a long and well thought out goal.

  - Preserves the draft tail for inline-argument slash commands like `/goal` and `/review` when completing with `Tab` or `Enter`.
  - Keeps popup filtering focused on the command fragment under the cursor rather than the full draft text.
  - Leaves slash commands that do not support inline arguments unchanged, so completion still replaces the existing draft tail for those commands.
  - Adds focused TUI tests under slash input covering preserved arguments, cursor edge cases, and the negative case for a command without inline args.
  
Follow-up simplification and test relocation from #24683 folded into this PR.

### CODE REVIEW

**Comment by canvrno-oai** [comment]
@codex review

File: codex-rs/tui/src/bottom_pane/chat_composer/slash_input.rs [hunk 1]
```diff
@@ -338,6 +380,69 @@ impl ChatComposer {
         }
     }
 
+    fn complete_selected_slash_command_preserving_existing_draft_tail_as_inline_args(
+        &mut self,
+        selected_cmd: &CommandItem,
+    ) -> bool {
```
[the remark this answers is not in the document]
Comment (canvrno-oai) [reply codex-rs/tui/src/bottom_pane/chat_composer/slash_input.rs:386]: Opening a separate draft PR to clean up the stale guidance and module-doc references:
https://github.com/openai/codex/pull/24641

**Review by etraut-openai — APPROVED** [review]
Some suggested changes that I made in a [stacked PR](https://github.com/openai/codex/pull/24683):
1. Simplify tests and move them out of chat_composer to keep that module from becoming obese again
2. Generalize to support all slash commands that accept arguments

Otherwise looks good. I tested it locally.

File: codex-rs/tui/src/bottom_pane/chat_composer/slash_input.rs [hunk 2]
```diff
@@ -267,8 +280,16 @@ impl ChatComposer {
...
+                        .complete_selected_slash_command_preserving_existing_draft_tail_as_inline_args(
+                            &selected_cmd,
+                        )
+                    {
+                        return (InputResult::None, true);
+                    }
+
                     if let Some(completed_text) =
-                        selected_command_completion(first_line, &selected_cmd)
+                        selected_command_completion(&first_line, &selected_cmd)
```
[the remark this answers is not in the document]
Comment (canvrno-oai) [reply codex-rs/tui/src/bottom_pane/chat_composer/slash_input.rs:292]: I'm accepting these minor cursor-position edge cases that require partially typed or combined command words, the scope is narrow.

File: codex-rs/tui/src/bottom_pane/chat_composer/slash_input.rs [hunk 3]
```diff
@@ -338,6 +380,71 @@ impl ChatComposer {
...
+        let typed_command_name = &text[1..command_token_end];
+        let rest_after_token_is_empty = text[command_token_end..].trim().is_empty();
+        if rest_after_token_is_empty && (cursor <= 1 || cursor >= command_token_end) {
+            return false;
+        }
+        let replace_end =
+            if cursor <= 1 || (typed_command_name == cmd.command() && rest_after_token_is_empty) {
+                command_token_end
+            } else {
+                cursor
```
[the remark this answers is not in the document]
Comment (canvrno-oai) [reply codex-rs/tui/src/bottom_pane/chat_composer/slash_input.rs:414]: I'm accepting these minor cursor-position edge cases that require partially typed or combined command words, the scope is narrow.
